### ISSUE TITLE
Set up ExploreASL VM for compiling and testing

### ISSUE THREAD

**Issue by HenkMutsaerts** [ExploreASL/ExploreASL#509]
We would like to move to a single ExploreASL VM that is used for 
A) compiling ExploreASL and building a Docker (or alternative)
B) running ExploreASL tests
(Eventually, this can then be automated by Github actions calling this VM with a specific shell script. Could also be a separate user, VM would be nicest to keep performance consistent over time.)

@pfcgroot & @MichaelStritt, which steps shall we define and in what order?
@pfcgroot I include you as this may get a bit more complicated, needing sudo access perhaps for some parts. 
If we have defined our wishes, steps to take and order, we can set up a call for this.

**Comment by MichaelStritt** [comment]
Puuuh... okay, full automation will require a lot of changes. I'm not really sure that this will be feasible for v1.7.0, because I need to work on some other PhD related things as well and the full automation is mostly a quality of life improvement...

This would technically require the following things:

1. A [GitHub action](https://github.com/features/actions) that will start with each main/develop merge. This should be the easiest step.
2. The GitHub action needs to call a [self-hosted runner](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) where Matlab is installed.
3. There we probably need a shell script which starts Matlab and runs the compiler.
4. Then the compilation needs to be stored and has to be accessible for the [Dockerfile](https://docs.docker.com/get-started/02_our_app/). So we basically need a web address for it like right now with google drive or we need to write something ssh related, but I don't know how this would work. Maybe the self-hosted runner can access these local files directly? I never tried something like this.
5. Then the GitHub action should push the updated (beta-) image to dockerhub like it already does right now.

**Comment by HenkMutsaerts** [comment]
Yes this is what I was thinking, but we could start in a simpler fashion without the automation, but then we at least have a dedicated testing environment. Although we may need to buy a compiler license...

**Comment by maartenhammer** [comment]
Change to functional account instead of virtual machine, deploy along with #1247 for automated testing.
